src/Storageless/Http/SessionMiddleware.php

```diff
@@ -224,36 +217,29 @@ private function appendToken(SessionInterface $sessionContainer, Response $respo
 
     private function shouldTokenBeRefreshed(?Token $token): bool
     {
-        if (! ($token && $token->hasClaim(self::ISSUED_AT_CLAIM))) {
-            return false;
-        }
-
-        $issuedAt = $token->getClaim(self::ISSUED_AT_CLAIM);
-
-        assert(is_numeric($issuedAt));
-
-        return $this->timestamp() >= $issuedAt + $this->refreshTime;
+        return $token !== null
+            && $token->hasBeenIssuedBefore($this->clock->now()->sub(new DateInterval(sprintf('PT%sS', $this->refreshTime))));
     }
 
     /**
      * @throws BadMethodCallException
      */
     private function getTokenCookie(SessionInterface $sessionContainer): SetCookie
     {
-        $timestamp = $this->timestamp();
+        $expiresAt = $this->clock->now()->add(new DateInterval(sprintf('PT%sS', $this->expirationTime)));
 
         return $this
             ->defaultCookie
             ->withValue(
-                (new Builder())
-                    ->setIssuedAt($timestamp)
-                    ->setExpiration($timestamp + $this->expirationTime)
-                    ->set(self::SESSION_CLAIM, $sessionContainer)
-                    ->sign($this->signer, $this->signatureKey)
-                    ->getToken()
-                    ->__toString()
+                $this->config->builder()
+                    ->issuedAt($this->clock->now())
+                    ->canOnlyBeUsedAfter($this->clock->now())
```

Comment:
`NOT_BEFORE` claim required by `StrictValidAt` token validator

Reply:
Should the `$this->clock->now()` result be reused for both `->issuedAt()` and `canOnlyBeUsedAfter()`? Would make sense to me, and it would lead to two identical values (good for compression algos)

Reply:
Done